torch/_dynamo/convert_frame.py

```diff
@@ -1027,26 +1026,12 @@ def format_guard_failures() -> str:
             from .utils import curr_frame
 
             frame_key = str(curr_frame)
-            if (
-                fail_reason is None
-                and output is not None
-                and frame_key in frame_phase_timing
-            ):
+            if fail_reason is None and output is not None:
                 guard_count = len(output.guards)
                 shape_env_guard_count = len(output.shape_env.guards)
                 graph_op_count = output.count_calls()
                 graph_node_count = len(output.graph.nodes)
                 graph_input_count = len(output.placeholders)
```

Comment:
The changes that follow in this file can be summed up by:
1) Rather than setting fields in a MetricsContext, we pass a dict of fields to the MetricsContext. It's the MetricsContext that will initiate the logging on exit.
2) Several fields are no longer updated here. They're set in the MetricsContext via dyanmo_timed.